jena-tdb2/src/main/java/org/apache/jena/tdb2/store/NodeIdFactory.java

```diff
@@ -102,7 +102,7 @@ private static NodeId create(int v1, long v2) {
         int t = v1 >> 24;
         NodeIdType type = NodeIdType.intToEnum(t);
         if ( type == NodeIdType.SPECIAL )
-            throw new TDBException(String.format("Attempt to create a special from a long: 0x%016", v2));
+            throw new TDBException(String.format("Attempt to create a special from a long: 0x%016X", v2));
```

Comment:
These two changes were the only ones I was not sure. Can't recall ever using this syntax. The rest of the changes look good!

Reply:
It's admittedly an odd one.  Without the X the format method doesn't know what it's doing and will throw an UnknownFormatConversionException. With the X it knows that it's hexadecimal, and the preceding numbers tell it how many characters to . 

I had to plug it into a wee online [java compiler](https://www.programiz.com/java-programming/online-compiler/) just to be sure.

`
public static void main(String[] args) {
        System.out.println(String.format("Will print a 16 character hex: 0x%016X", 13L));
        System.out.println(String.format("Will throw an exception: 0x%016", 13L));
    }
`